### PR TITLE
better client logs for flaky insert and work test, shorter test connect timeout

### PR DESCRIPTION
**Pull request by bgentry**
This contains a few small fixes and logging improvements encountered while debugging flaky tests from #213. The main one that actually fixes one of the flaky scenarios is included in the final commit: shortening `ConnectTimeout` from 10s to 2s in tests. By failing faster than our typical test timeout, the occasional flaky conn attempt that hangs will no longer make the test immediately fail. In my testing this completely eliminated this one flaky test cause (of several).

### CODE REVIEW

**Comment by bgentry** [comment]
@brandur this can of course wait until after #212 is merged. Don't want to delay that one at all, and it should be easy to rebase this little on / figure out what equivalent changes to make once that is merged.

**Review by brandur — APPROVED** [review]

**Pull request merged**
